RevenueCatUI/View+PresentPaywall.swift

```diff
@@ -560,14 +681,299 @@ private struct PresentingPaywallModifier: ViewModifier {
+                        #if targetEnvironment(macCatalyst) || os(macOS)
+                            .frame(minHeight: 667)
+                        #endif
+                    }
+                    .sheet(item: self.$presentedExitOffer, onDismiss: self.handleExitOfferDismiss) { exitOffering in
+                        self.exitOfferPaywallView(for: exitOffering)
+                        #if targetEnvironment(macCatalyst) || os(macOS)
+                            .frame(minHeight: 667)
+                        #endif
+                    }
+            #if !os(macOS)
+            case .fullScreen:
+                content
+                    .fullScreenCover(item: self.$offering, onDismiss: self.handleMainPaywallDismiss) { offering in
+                        self.paywallView(for: offering)
+                    }
+                    .fullScreenCover(
+                        item: self.$presentedExitOffer,
+                        onDismiss: self.handleExitOfferDismiss
+                    ) { exitOffering in
+                        self.exitOfferPaywallView(for: exitOffering)
+                    }
+            #endif
+            }
+        }
+    }
+
+    private func paywallView(for offering: Offering) -> some View {
+        PaywallView(
+            configuration: .init(
+                content: .offering(offering),
+                fonts: self.fontProvider,
+                displayCloseButton: true,
+                purchaseHandler: self.purchaseHandler
+            )
+        )
+        .withEventHandlers(
+            purchaseHandler: self.purchaseHandler,
+            purchaseStarted: self.purchaseStarted,
+            purchaseCompleted: self.purchaseCompleted,
+            purchaseCancelled: self.purchaseCancelled,
+            restoreStarted: self.restoreStarted,
+            restoreCompleted: self.restoreCompleted,
+            purchaseFailure: self.purchaseFailure,
+            restoreFailure: self.restoreFailure
+        )
+        .task(id: offering.identifier) {
+            // Reset session purchase result to track purchases in THIS session only.
+            // Using .task(id:) ensures this only runs once per unique offering,
+            // avoiding accidental resets if onAppear is called multiple times.
+            self.purchaseHandler.resetSessionPurchaseResult()
+
+            self.exitOfferOffering = await ExitOfferHelper.fetchValidExitOffer(for: offering)
+        }
+    }
+
+    private func exitOfferPaywallView(for offering: Offering) -> some View {
+        PaywallView(
+            configuration: .init(
+                content: .offering(offering),
+                fonts: self.fontProvider,
+                displayCloseButton: true,
+                purchaseHandler: self.purchaseHandler
+            )
+        )
+        .withEventHandlers(
+            purchaseHandler: self.purchaseHandler,
+            purchaseStarted: self.purchaseStarted,
+            purchaseCompleted: self.purchaseCompleted,
```

Comment:
should this `purchaseCompleted` also call `close`? The other modifier has calls to close the paywall in `purchaseCompleted` and in `RestoredCustomerInfoPreferenceKey` changes. Do we need that? Or what will close the paywall on purchase completed?